### DISCUSSION TITLE
Data Migration method

### DISCUSSION THREAD

**Discussion by bingbb47** [milvus-io/milvus#20646] [Q&A and General discussion] [answered]
as the title, how to migrate a collection to another cluster? i have try to query the whole collection and insert , but there is no iterative query method;
the Data migration tools in milvus-tools can only read meta in mysql;
did you have any idea about how to migrate data with ETCD

**Comment by yhmo** [comment, marked as answer]
We are building a buckup tool for milvus. The tool can extract vector data(not including index data) and metadata from a milvus cluster, then import the vectors into another milvus cluster. Users can specify target collection to export/import.

**Reply by bingbb47** [reply]
Expect milvus 2.2 ！

**Reply by simonaries** [reply]
How soon will the buckup be ready? Is it open source? You can now find persistent data stored on minio, but there is no way to deserialize it

**Reply by yhmo** [reply]
The backup tool is here: https://github.com/zilliztech/milvus-backup
Now it is in Beta stage. Read the README to know how to use it: https://github.com/zilliztech/milvus-backup/blob/master/README.md

Note: this tool only exports the original data, not including the index, you need to create index after restoring the data on another milvus(v2.2) cluster.
